src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -85,7 +85,10 @@ protected ApiConfiguration Configuration
                     apiType =>
                     {
                         IServiceCollection services = new ServiceCollection()
-                            .CutoffPrevious<IQueryExecutor>(DefaultQueryExecutor.Instance);
+                            .CutoffPrevious<IQueryExecutor>(DefaultQueryExecutor.Instance)
+                            .AddSingleton<ApiConfigurationExtensions.PropertyBag>()
+                            .AddScoped<ApiContextExtensions.PropertyBag>()
+                            .AddScoped<InvocationContextExtensions.PropertyBag>();
```

Comment:
Yeah exactly, root provider is also in a scope, although without ServiceScope object. Actually we should even dispose it if under some circumstances, we're done with an ApiConfiguration. The way is `((IDisposable)serviceProvider).Dispose()`